torch/csrc/autograd/record_function.h

```diff
@@ -134,6 +134,14 @@ struct TORCH_API RecordFunction {
   // that it was not set with setThreadId() and this RecordFunction's callbacks
   // cannot be invoked from a separate thread.
   uint16_t threadId_ = 0;
+  // Tracks the state of the RecordFunction through its lifetime. Used for error
+  // handling and ensuring that users utilize RecordFunction's functions in the
+  // right way.
+  unsigned int state = 0;
+  static constexpr int BEFORE_CALLBACKS_RAN = 1 << 0;
```

Comment:
should this be the same type (`unsigned`) as `state` as the code performs bit operations on them?